### PR TITLE
Feat: Created DNS module

### PR DESCRIPTION
**Pull request by Tanveer143s**
## what
* Created GCP DNS zone module

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ updates:
     open-pull-requests-limit: 3
 
   - package-ecosystem: "terraform" # See documentation for possible values
-    directory: "/_example/basic" # Location of package manifests
+    directory: "/_example/peering-zone" # Location of package manifests
```
Comment (VishwajitNagulkar) [inline .github/dependabot.yml]: Update directory to `examples`

File: .github/workflows/readme.yml [hunk 2]
```diff
@@ -51,4 +50,4 @@ jobs:
         env:
```
Comment (VishwajitNagulkar) [inline .github/workflows/readme.yml]: use shared workflow for readme

File: README.yaml [hunk 3]
```diff
@@ -11,34 +11,88 @@ name : Terraform Module Template
...
+    url: "https://github.com/clouddrove/terraform-gcp-dns/actions/workflows/terraform.yml"
 
-prerequesties:
-  - name: Terraform 1.4.6
-    url: https://learn.hashicorp.com/terraform/getting-started/install.html
-    
 #  description of this project
 description: |-
-  Terraform module template to create new modules using this as baseline
+  Terraform module to create subnet resource on google.
```
Comment (VishwajitNagulkar) [inline README.yaml]: Update description according to DNS

File: README.yaml [hunk 4]
```diff
@@ -11,34 +11,88 @@ name : Terraform Module Template
...
 # extra content
 include:
   - "terraform.md"
```
Comment (VishwajitNagulkar) [inline README.yaml]: remove this

File: README.yaml [hunk 5]
```diff
@@ -11,34 +11,88 @@ name : Terraform Module Template
...
 
-# How to use this project
-# How to use this project
+
 usage: |-
   Here are some examples of how you can use this module in your inventory structure:
+  ### Default subnet
```
Comment (VishwajitNagulkar) [inline README.yaml]: Update comment according to dns

File: examples/private-zone/main.tf [hunk 6]
```diff
@@ -0,0 +1,57 @@
+
+module "dns-private-zone" {
+  source = "../.."
+  # version = "~> 5.0"
```
Comment (VishwajitNagulkar) [inline examples/private-zone/main.tf]: remove that comment

File: .github/workflows/readme.yml (hunk 2 above)
Comment (Tanveer143s) [reply .github/workflows/readme.yml]: done

File: .github/dependabot.yml (hunk 1 above)
Comment (Tanveer143s) [reply .github/dependabot.yml]: done

File: README.yaml (hunk 3 above)
Comment (Tanveer143s) [reply README.yaml]: changed

File: README.yaml (hunk 5 above)
Comment (Tanveer143s) [reply README.yaml]: done

File: README.yaml (hunk 4 above)
Comment (Tanveer143s) [reply README.yaml]: done

File: examples/private-zone/main.tf (hunk 6 above)
Comment (Tanveer143s) [reply examples/private-zone/main.tf]: done

**Review by VishwajitNagulkar — APPROVED** [review]
LGTM

**Review by VishwajitNagulkar — APPROVED** [review]

**Review by VishwajitNagulkar — APPROVED** [review]

**Pull request merged**